### PR TITLE
[Upgrade Watcher] Add logging to rollback restart step

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR adds logging to the restart step of the Agent upgrade rollback process.

## Why is it important?

To better understand the progress of the upgrade rollback process.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have made corresponding change to the default configuration files~
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] ~I have added an integration test or an E2E test~

## How to test this PR locally

1. Build an install Agent from `main` (currently at version `8.11.0`).
   ```
   EXTERNAL=true SNAPSHOT=true  DEV=true PLATFORMS=linux/arm64 PACKAGES=tar.gz mage package
   ```
3. Build Agent package from this PR (currently at version `8.12.0`).  Make sure NOT to use `SNAPSHOT=true` or `EXTERNAL=true`.
   ```
   DEV=true PLATFORMS=linux/arm64 PACKAGES=tar.gz mage package
   ```
4. Upgrade the installed `8.11.0` Agent to the `8.12.0` Agent built with this PR.
   ```
   sudo elastic-agent upgrade 8.12.0 --source-uri file://$(pwd)/build/distributions/ --skip-verify
   ```
6. Tail the Upgrade Watcher logs and ensure that the log messages in this PR show up.
   ```
   tail -F /opt/Elastic/Agent/data/elastic-agent-b8da42/logs/elastic-agent-watcher-$(date +%Y%m%d).ndjson
   ```

### CODE REVIEW

File: version/version.go
Comment (ycombinator) [inline version/version.go:8]: Will remove this after PR is tested and approved.

**Comment by cmacknz** [comment]
While you are in here, can you add calls to the `Sync` method of our logger before the watcher exits? I think the agent might be missing it as well since I don't see calls to it in the agent code directly.

File: version/version.go
Comment (ycombinator) [inline version/version.go:8]: Will remove this after PR is tested and approved.

File: internal/pkg/agent/cmd/run.go
Comment (cmacknz) [inline internal/pkg/agent/cmd/run.go:156]: Wouldn't it be safer to do this on `baseLogger` in case that ends up being passed somewhere else?

**Review by cmacknz — APPROVED** [review]
LGTM assuming you remember to remove the test code you added :p

One nit about which logger gets Sync'd in the agent.

File: internal/pkg/agent/cmd/run.go
Comment (ycombinator) [reply internal/pkg/agent/cmd/run.go]: Done in 367dfbf41.

**Comment by ycombinator** [comment]
buildkite test this

File: internal/pkg/agent/application/upgrade/rollback.go
Comment (michalpristas) [inline internal/pkg/agent/application/upgrade/rollback.go:151]: wouldnt it be easier and more optimal to switch for logic and condition on line 159?

**Review by michalpristas — APPROVED** [review]

File: internal/pkg/agent/application/upgrade/rollback.go
Comment (ycombinator) [reply internal/pkg/agent/application/upgrade/rollback.go:151]: You mean to make the loop go from 0 to `maxRestartCount` instead of how it is now, going in the opposite direction?

Comment (ycombinator) [reply internal/pkg/agent/application/upgrade/rollback.go]: @michalpristas I made a change to the logic in 5ec79d4ca. Is this what you were suggesting?

Comment (michalpristas) [reply internal/pkg/agent/application/upgrade/rollback.go]: yeah, much nicer and no arithmetic operations on top of the loop

**Review by pchila — APPROVED** [review]

**Pull request merged**
